docs/migration/migrate_6_0_index.asciidoc

```diff
@@ -0,0 +1,337 @@
+[[migrating-6.0-index]]
+== Migrating Kibana index to 6.0
+
+[IMPORTANT]
+==============================================
+The following will make transformations to your Kibana index. It is recommended you backup your Kibana index prior to running any of the following commands.
+==============================================
+
+In order to migrate to Kibana 6, your Kibana index needs to be re-indexed. This will make transformations to the data model, removing types which are being {es-ref}/removal-of-types.html[removed in Elasticsearch].
```

Comment:
I think in other docs, we aren't hyphenating "reindex", so maybe being consistent here would be good.